src/AWS.Deploy.CLI/Commands/DeployCommand.cs

```diff
@@ -194,6 +201,129 @@ public async Task ExecuteAsync(string stackName, bool saveCdkProject)
             await orchestrator.DeployRecommendation(cloudApplication, selectedRecommendation);
         }
 
+        /// <summary>
+        /// This method is used to set the values for Option Setting Items when a deployment is being performed using a user specifed config file.
+        /// </summary>
+        /// <param name="recommendation">The selected recommendation settings used for deployment <see cref="Recommendation"/></param>
+        /// <param name="deploymentBundleDefinition">The container for the deployment bundle used by an application. <see cref="DeploymentBundleDefinition"/></param>
+        /// <param name="userDeploymentSettings">The deserialized object from the user provided config file. <see cref="UserDeploymentSettings"></param>
+        private async Task ConfigureDeployment(Recommendation recommendation, DeploymentBundleDefinition deploymentBundleDefinition, UserDeploymentSettings userDeploymentSettings)
+        {
+            foreach (var entry in userDeploymentSettings.OptionSettingItems)
+            {
+                var optionSettingJsonPath = entry.Key;
+                var optionSettingValue = entry.Value;
+
+                var isPartOfDeploymentBundle = true;
+                var optionSetting = deploymentBundleDefinition.Parameters.FirstOrDefault(x => x.Id.Equals(optionSettingJsonPath));
+                if (optionSetting == null)
+                {
+                    optionSetting = recommendation.GetOptionSetting(optionSettingJsonPath);
+                    isPartOfDeploymentBundle = false;
+                }
+                    
+                if (!recommendation.IsExistingCloudApplication || optionSetting.Updatable)
+                {
+                    object settingValue;
+                    try
+                    {
+                        switch (optionSetting.Type)

```

Comment:
I understand now that when `UserDeploymentSettings` is deserialized it is pruning the option settings to the leaf nodes. Overall I'm not a fan of this approach because it breaks the separation of concerns. Right now `UserDeploymentSettings` has some pretty weird logic of rearranging the data structures for the data in the JSON file just to match what this function needs. It makes the code confusing and hard to follow.

Reply:
The Option Setting type that goes inside the switch case, will always be of a primitive type because it is a leaf Option Setting Item.